### PR TITLE
Modify GroupView UI

### PR DESCRIPTION
**Pull request by lukebarnard1**
- Remove featured users/rooms
 - Add "Rooms" section to show all rooms in the group in a room-directory-esque list. This has a "+" button in "edit" mode.
 - Make the group view body scrollable

### CODE REVIEW

File: src/components/structures/GroupView.js [hunk 1]
```diff
@@ -935,7 +967,12 @@ export default React.createClass({
                             { rightButtons }
                         </div>
                     </div>
-                    { roomBody }
+                    <ScrollPanel className="mx_GroupView_body"
```
Comment (dbkr) [inline src/components/structures/GroupView.js]: ScrollPanel might be overkill here as it's designed for timeline style infinite pagination things - in UserSettings we just put a GeminiScrollbar in, for instance.

File: src/components/structures/GroupView.js [hunk 2]
```diff
@@ -856,14 +888,15 @@ export default React.createClass({
...
-                roomBody = <div>
-                    <textarea className="mx_GroupView_editLongDesc" value={this.state.profileForm.long_description}
+                bodyNodes = [
+                    <textarea className="mx_GroupView_editLongDesc"
+                        value={this.state.profileForm.long_description}
                         onChange={this._onLongDescChange}
                         tabIndex="3"
-                    />
-                    { this._getFeaturedRoomsNode() }
-                    { this._getFeaturedUsersNode() }
```
Comment (dbkr) [inline src/components/structures/GroupView.js]: So featured rooms / users are unused now? Should we remove the functions?

File: src/components/views/rooms/RoomDetailList.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    render: function() {
+        const BaseAvatar = sdk.getComponent('avatars.BaseAvatar');
+
+        const room = this.props.room;
+        const name = room.name || getDisplayAliasForRoom(room) || _t('Unnamed room');
+        const topic = linkifyString(sanitizeHtml(room.topic || ''));
+
+        const guestRead = room.world_readable ? (
+                <div className="mx_RoomDirectory_perm">{ _t('World readable') }</div>
```
Comment (dbkr) [inline src/components/views/rooms/RoomDetailList.js]: I guess this should share code with the public room directory, at least in the long term

File: src/components/views/rooms/RoomDetailList.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            ) : <div />;
+
+        const perms = (guestRead || guestJoin) ? (<div className="mx_RoomDirectory_perms">
+            { guestRead }
+            { guestJoin }
+        </div>) : <div />;
+
+        return <tr key={room.room_id} onClick={this.onClick}>
+            <td className="mx_RoomDirectory_roomAvatar">
+                <BaseAvatar width={24} height={24} resizeMethod='crop'
```
Comment (dbkr) [inline src/components/views/rooms/RoomDetailList.js]: RoomAvatar?

File: src/components/views/rooms/RoomDetailList.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                    name={name} idName={name}
+                    url={ContentRepo.getHttpUriForMxc(
+                            MatrixClientPeg.get().getHomeserverUrl(),
+                            room.avatar_url, 24, 24, "crop")} />
+            </td>
+            <td className="mx_RoomDirectory_roomDescription">
+                <div className="mx_RoomDirectory_name">{ name }</div>&nbsp;
+                { perms }
+                <div className="mx_RoomDirectory_topic"
+                     onClick={function(e) { e.stopPropagation(); }}
```
Comment (dbkr) [inline src/components/views/rooms/RoomDetailList.js]: What's this for? (It's also making a new function on each render so will break react's DOM diffing)

File: src/components/structures/GroupView.js (hunk 1 above)
Comment (lukebarnard1) [reply src/components/structures/GroupView.js]: Ah, that would make sense

File: src/components/views/rooms/RoomDetailList.js (hunk 3 above)
Comment (lukebarnard1) [reply src/components/views/rooms/RoomDetailList.js]: Sure, in the long term. It was too entrenched in RoomDirectory so I decided not to do that yet.

File: src/components/structures/GroupView.js (hunk 2 above)
Comment (lukebarnard1) [reply src/components/structures/GroupView.js]: I can't really justify keeping them. Although, we're going to need them Eventually...

File: src/components/views/rooms/RoomDetailList.js (hunk 4 above)
Comment (lukebarnard1) [reply src/components/views/rooms/RoomDetailList.js]: `room` isn't actually a `Room` object here

Comment (dbkr) [reply src/components/views/rooms/RoomDetailList.js]: oh, of course

File: src/components/structures/ScrollPanel.js [hunk 6]
```diff
@@ -148,6 +148,7 @@ module.exports = React.createClass({
             onFillRequest: function(backwards) { return Promise.resolve(false); },
             onUnfillRequest: function(backwards, scrollToken) {},
             onScroll: function() {},
+            onResize: function() {},
```
Comment (dbkr) [inline src/components/structures/ScrollPanel.js]: Is this still useful to add? Seems like it might be

Comment (lukebarnard1) [reply src/components/structures/ScrollPanel.js]: Oh. I had left it in unintentionally. I think it is useful; things might not need to do anything if the SP changes size.

**Pull request merged**